docs/http/http-spans.md

```diff
@@ -226,33 +226,97 @@ For an HTTP client span, `SpanKind` MUST be `Client`.
+**[5]:** If the request fails with an error before response status code was sent or received,
+`error.type` SHOULD be set to exception type (its fully-qualified class name, if applicable)
+or a component-specific low cardinality error identifier.
+
+If response status code was sent or received and status indicates an error according to [HTTP span status definition](/docs/http/http-spans.md),
+`error.type` SHOULD be set to the status code number (represented as a string), an exception type (if thrown) or a component-specific error identifier.
+
+The `error.type` value SHOULD be predictable and SHOULD have low cardinality.
+Instrumentations SHOULD document the list of errors they report.
 
-**[5]:** Instrumentations SHOULD require an explicit configuration of which headers are to be captured. Including all request headers can be a security risk - explicit configuration helps avoid leaking sensitive information.
+The cardinality of `error.type` within one instrumentation library SHOULD be low, but
+telemetry consumers that aggregate data from multiple instrumentation libraries and applications
+should be prepared for `error.type` to have high cardinality at query time, when no
+additional filters are applied.
+
+If the request has completed successfully, instrumentations SHOULD NOT set `error.type`.
+
+**[6]:** The value SHOULD be normalized to lowercase.
+
+**[7]:** If not `http` and `network.protocol.version` is set.
+
+**[8]:** The resend count SHOULD be updated each time an HTTP request gets resent by the client, regardless of what was the cause of the resending (e.g. redirection, authorization failure, 503 Server Unavailable, network issues, or any other).
+
+**[9]:** If protocol version is subject to negotiation (for example using [ALPN](https://www.rfc-editor.org/rfc/rfc7301.html)), this attribute SHOULD be set to the negotiated version. If the actual protocol version is not known, this attribute SHOULD NOT be set.
+
+**[10]:** Instrumentations SHOULD require an explicit configuration of which headers are to be captured. Including all request headers can be a security risk - explicit configuration helps avoid leaking sensitive information.
 The `User-Agent` header is already captured in the `user_agent.original` attribute. Users MAY explicitly configure instrumentations to capture them even though it is not recommended.
 The attribute value MUST consist of either multiple header values as an array of strings or a single-item array containing a possibly comma-concatenated string, depending on the way the HTTP library provides access to headers.
 
 The following attributes can be important for making sampling decisions and SHOULD be provided **at span creation time** (if provided at all):
 
-* [`server.address`](../attributes-registry/server.md)
-* [`server.port`](../attributes-registry/server.md)
-* [`url.full`](../attributes-registry/url.md)
+* [`server.address`](/docs/attributes-registry/server.md)
```

Comment:
`http.request.method` is missing here - I think there is a bug in finding all sampling-relevant attributes (which probably came from the build-tools).

It became obvious now when we listed all attributes

Reply:
I figured it out.

There is *NO* link from `trace.http.common` to either `trace.http.server` or `trace.http.client`.

This bug *would* show up in codegen if we did more than generate attribute constants.

Reply:
See: https://github.com/open-telemetry/semantic-conventions/issues/928

Weaver is working correctly....